### ISSUE TITLE
Alternative architectures

### ISSUE THREAD

**Issue by hbredin** [pyannote/pyannote-audio#51]
- [ ] [Speaker Diarization using Deep Recurrent Convolutional Neural Networks for Speaker Embeddings](https://arxiv.org/pdf/1708.02840.pdf)
- [ ] [End-to-end text-independent speaker verification with flexibility in utterance duration](https://www.researchgate.net/publication/320011502_END-TO-END_TEXT-INDEPENDENT_SPEAKER_VERIFICATION_WITH_FLEXIBILITY_IN_UTTERANCE_DURATION)
- [ ] [VoxCeleb: a large-scale speaker identification dataset](http://www.robots.ox.ac.uk/~vgg/data/voxceleb/)
- [ ] [An Empirical Evaluation of Generic Convolutional and Recurrent Networks for Sequence Modeling](https://github.com/locuslab/TCN)
- [ ] [Analysis of Length Normalization in End-to-End Speaker Verification System](https://arxiv.org/pdf/1806.03209.pdf)  
Authors show that normalizing embeddings on a larger sphere leads to better results. 
- [ ] [Speaker Recognition from Raw Waveform with SincNet](https://arxiv.org/abs/1808.00158)  
- [ ] [Adaptive pooling operators for weakly labeled sound event detection](https://github.com/marl/autopool)  
This could be used to learn the final pooling layer in LSTM-based embedding 
- [x] [TDNN](https://github.com/SiddGururani/Pytorch-TDNN)

**Comment by mcompute** [comment]
Awesome toolkit and excited to experiment on it using my own training dataset! :-)

Have you also considered single neural building block to solving speaker diarization problem as mentioned in this [paper - End-to-End Neural Diarization: Reformulating Speaker Diarization as Simple Multi-label Classification](https://arxiv.org/pdf/2003.02966.pdf)?
Self-attention model was used and has shown comparable/better result than BLSTM and x-vector clustering methods. 
The team has provided code on [github](https://github.com/hitachi-speech/EEND) to reproduce their experiments.

**Comment by hbredin** [comment]
No, I haven't considered this approach because, as far as I know, it has only been used for two-speakers conversations.

**Comment by mcompute** [comment]
Paper author has mentioned about performance on larger speakers set in **_Results section E "Effect of varying number of heads in self-attention blocks"_**. 
I guessed author has not stated the datasets experimented to produce the larger speakers set results.

**Comment by hbredin** [comment]
I am sorry but I think you misunderstood what Section E is about: it is about the number of transformer heads, not the number of speakers in the diarized conversation. I suggest you contact the authors directly as I am not, in any way, related to this work.
